### DISCUSSION TITLE
how to delete records from the stream using custom field

### DISCUSSION THREAD

**Discussion by ankitsoin** [finos/TimeBase-CE#249] [Q&A]
I want to delete records from a space based on some batch id using python.
(My space contains multiple batch data, ask is to overwrite the existing data by removing the existing batch id.)

Can you please help with the query or what approach should we take?

Thanks,
Ankit

**Comment by ankitsoin** [comment]
Also, if I see this 'Delete' Section on this https://kb.timebase.info/community/technology/apis/tickstream#delete
I cant find any python implementation.
Is it available for python?

**Reply by alex-karpovich** [reply]
Greetings, @ankitsoin 
> I want to delete records from a space based on some batch id using python.

Since TimeBase is time-series database and it's has specific storage designed for it, we do not support deleting data by some conditions
like classical SQL `delete from .. where id == <value>`
We support 'deletes' based on time ranges only.

What I can recommend - you can write simple python script which will read all data from specific space filtering using batch ID and write to another space. And then delete source space.

**Reply by ankitsoin** [reply]
Thanks @alex-karpovich for your reply.

So to delete using list of symbol and timestamp(from , to) using python implementation is not currently present? is it
Also I want to understand the power of 'spaces' , can single stream can handle 5000 or even more spaces for 3 -4 months or so (around 50 spaces for a day) without performance impact?

can you please suggest

**Reply by alex-karpovich** [reply]
@ankitsoin Correct, python API doesn't support delete(from, to) right now.

About spaces: they are generally like 'partitions' in other databases, because has independent storage.
Mostly common usage of spaces - parallel writing with different instruments or different time ranges.
Another usage is complex data processing which requires parallel data transformations and grouping data by different categories

Recommend usage of spaces: 
 - data partitioning by time ranges, say each space define one day of data
 - data partitioning by instruments, say each space contains independent number of symbols. 

Most of cases assuming that historical data may be changed and 'spaces' provide more `flexible` way of changing data.


Please describe how do you distribute data between spaces.
btw, TimeBase also has INSERT mode to insert historical data
